moshi/src/main/java/com/squareup/moshi/ClassFactory.java

```diff
@@ -54,29 +54,43 @@ abstract T newInstance() throws
       // No no-args constructor. Fall back to something more magical...
     }
 
-    // Try the JVM's Unsafe mechanism.
-    // public class Unsafe {
-    //   public Object allocateInstance(Class<?> type);
-    // }
+    boolean tryUnsafe = true;
```

Comment:
Make it a static field?

Reply:
Would we need to make it nullable to know if it's been set or not?